tools/parity_scale_decoder/src/generate_map.rs

```diff
@@ -0,0 +1,386 @@
+//! Exports `generate_map()` function and contains implementation details for it
+
+use std::collections::BTreeSet;
+
+use iroha_core::*;
+use iroha_crypto::*;
+use iroha_data_model::{prelude::*, *};
+use iroha_schema::IntoSchema;
+use iroha_version::*;
+
+use super::*;
+
+/// Trait to retrieve type name
+///
+/// It is used with abusing [inherit impls](https://doc.rust-lang.org/reference/items/implementations.html#inherent-implementations)
+/// to get `None` variant from types, which doesn't implement [`IntoSchema`] and `Some` which does
+trait TypeName {
+    /// Get name of the type or `None` if type doesn't implement `IntoSchema`
+    fn type_name() -> Option<String>;
+}
+
+impl<T> TypeName for T {
+    fn type_name() -> Option<String> {
+        None
+    }
+}
+
+/// Neotype which has `type_name()` method when `T` implements [`IntoSchema`]
+struct WithTypeName<T>(std::marker::PhantomData<T>);
+
+impl<T: IntoSchema> WithTypeName<T> {
```

Comment:
```suggestion
impl<T: IntoSchema + Decode> WithTypeName<T> {
```

How about this? You said some don't implement `Decode`